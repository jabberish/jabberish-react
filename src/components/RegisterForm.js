import React from 'react';

const RegisterForm = () => {
  return (
    <form id="register-form">
      <label>username:
        <input id="username" name="username" required />
      </label>
      <label>Password:
        <input id="password" name="password" type="password" required />
      </label>
      <button>Submit</button>
    </form>
  );
};

export default RegisterForm;
