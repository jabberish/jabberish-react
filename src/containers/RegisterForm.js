import React from 'react';
import PropTypes from 'prop-types';

class RegisterForm extends React.Component {
  state = {
    username: '',
    password: ''
  }

  onSubmit = e => {
    e.preventDefault();
    this.props.submitRegister({ username: this.state.username, password: this.state.password });
  };

  onUsernameChange = e => {
    this.setState({ username: e.target.value });
  }

  onPasswordChange = e => {
    this.setState({ password: e.target.value });
  }

  render() {
    return (
      <form id="register-form">
        <label>username:
          <input 
            id="username" 
            name="username" 
            onChange={this.onUsernameChange}
            required 
          />
        </label>
        <label>Password:
          <input 
            id="password" 
            name="password" 
            type="password" 
            onChange={this.onPasswordChange}
            required 
          />
        </label>
        <button onClick={this.onSubmit}>Submit</button>
      </form>
    );
  }
}

RegisterForm.propTypes = {
  submitRegister: PropTypes.func.isRequired
};

export default RegisterForm;
