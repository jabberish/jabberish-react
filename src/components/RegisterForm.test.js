import React from 'react';
import { shallow } from 'enzyme';
import RegisterForm from './RegisterForm';

describe('RegisterForm component', () => {
  it('renders RegisterForm', () => {
    const wrapper = shallow(<RegisterForm />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
