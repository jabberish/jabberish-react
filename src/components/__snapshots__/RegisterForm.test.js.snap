// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm component renders RegisterForm 1`] = `
<form
  id="register-form"
>
  <label>
    username:
    <input
      id="username"
      name="username"
      required={true}
    />
  </label>
  <label>
    Password:
    <input
      id="password"
      name="password"
      required={true}
      type="password"
    />
  </label>
  <button>
    Submit
  </button>
</form>
`;
